### ISSUE TITLE
Question : override defaultErrorBuilder globaly

### ISSUE THREAD

**Issue by sm2017** [GiancarloCode/form_bloc#6]
I want to override `defaultErrorBuilder` to customize error messages globaly , How can I do this?

I see that each fields accept `errorBuilder` passed to `Style.getErrorText` , it is very cumbersome to pass my errorBuilder each time to each fields , How can I do it easily?

I think we can extend `TextFieldBlocBuilder` and create new constructor , but I dont know can can I do it in dart in clean way

**Comment by sm2017** [comment]
Any idea?

**Comment by GiancarloCode** [comment]
@sm2017 Hello
As I understand you want to create a unique function that is responsible for managing the mapping / translation of the error codes of all forms, and this function you only want to pass it once.

If not, please correct me.

The first solution is to create a new widget which has defined `errorBuilder`, but it is quite repetitive since you will have to create every property you want to use, but this is how you should do it t in dart when work with widgets.

```dart
class MyCustomTextFieldBlocBuilder extends StatelessWidget {
  final TextFieldBloc textFieldBloc;
  // Other fields...

  const MyCustomTextFieldBlocBuilder({
    Key key,
    @required this.textFieldBloc,
  }) : super(key: key);

  String _errorBuilder(BuildContext context, String error) {
    // Awesome logic...
    return error;
  }

  @override
  Widget build(BuildContext context) {
    return TextFieldBlocBuilder(
      textFieldBloc: textFieldBloc,
      errorBuilder: _errorBuilder,
    );
  }
}
```

___


Also I released `flutter_form_bloc: ^0.6.1-beta-1`,  I just created the `FieldBlocBuilder` class with the static property ` defaultErrorBuilder`.

So now you can overwrite this property with the function you want at the beginning of your program, please try it and tell me if it adapts to your use case.

```dart
void main() {
  ...
  FieldBlocBuilder.defaultErrorBuilder = (BuildContext context, String error) {
    return 'Awesome error';
  };

  runApp(App());
}
```



And if you have any improvement or suggestion please let me know.

**Comment by sm2017** [comment]
@GiancarloCode thank you very much

**Comment by NestorDevs** [comment]
> @sm2017 Hello
> As I understand you want to create a unique function that is responsible for managing the mapping / translation of the error codes of all forms, and this function you only want to pass it once.
> 
> If not, please correct me.
> 
> The first solution is to create a new widget which has defined `errorBuilder`, but it is quite repetitive since you will have to create every property you want to use, but this is how you should do it t in dart when work with widgets.
> 
> ```dart
> class MyCustomTextFieldBlocBuilder extends StatelessWidget {
>   final TextFieldBloc textFieldBloc;
>   // Other fields...
> 
>   const MyCustomTextFieldBlocBuilder({
>     Key key,
>     @required this.textFieldBloc,
>   }) : super(key: key);
> 
>   String _errorBuilder(BuildContext context, String error) {
>     // Awesome logic...
>     return error;
>   }
> 
>   @override
>   Widget build(BuildContext context) {
>     return TextFieldBlocBuilder(
>       textFieldBloc: textFieldBloc,
>       errorBuilder: _errorBuilder,
>     );
>   }
> }
> ```
> 
> Also I released `flutter_form_bloc: ^0.6.1-beta-1`, I just created the `FieldBlocBuilder` class with the static property ` defaultErrorBuilder`.
> 
> So now you can overwrite this property with the function you want at the beginning of your program, please try it and tell me if it adapts to your use case.
> 
> ```dart
> void main() {
>   ...
>   FieldBlocBuilder.defaultErrorBuilder = (BuildContext context, String error) {
>     return 'Awesome error';
>   };
> 
>   runApp(App());
> }
> ```
> 
> And if you have any improvement or suggestion please let me know.

pero aqui esta funciona va en el main? la verdad no entiendo puedes explicarme un poco mas aqui?
